### ISSUE TITLE
ListView fails to restore focus to dropped item after DnD reorder

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#4258]
# 🐛 Bug Report 

The issue does not appear in Storybook example, but can be reproduced in the docs:

1. Open https://react-spectrum.adobe.com/react-spectrum/ListView.html#reorderable
2. Tab to navigate to the ListView within the Reorderable example.
3. Press ArrowRight to move focus to the drag handle for the ListView item that has focus.
4. Press Enter to activate the drag handle and start reorder operation.
5. Use ArrowDown or ArrowUp to select a new insert position to drop the selected item.
6. Press Enter to drop the item in the new position.

## 🤔 Expected Behavior

7. Keyboard focus should be restored to the item that was just reordered. 

Note that the expected behavior seems to be reproduced in the Storybook example: https://reactspectrum.blob.core.windows.net/reactspectrum/1fc4185c117742e52337431ae8fbb0c392ee72da/storybook/index.html?path=/story/listview-drag-and-drop--drag-within-list-reorder&providerSwitcher-express=false

## 😯 Current Behavior

7. Keyboard focus restores to the ListView component itself.

## 💁 Possible Solution

Possibly resolved as part of #4248, specifically: https://github.com/adobe/react-spectrum/pull/4248/commits/b108520a50d3db06c11785730ffd5a646d01fef7

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | @adobe/react-spectrum@3.25.1
| Browser          | Chrome Version 111.0.5563.64 (Official Build) (arm64)
| Operating System | macOS Ventura 13.2.1 (22D68)

## 🧢 Your Company/Team

Adobe/Accessibility

## 🕷 Tracking Issue (optional)

 #4248

**Comment by LFDanLu** [comment]
Looks like this has been addressed, I can no longer reproduce it in the docs. Can you double confirm?
